lib/services/project-data-service.ts

```diff
@@ -1,74 +1,124 @@
+			try {
+				return this.getPropertyValueFromJson(projectData, propertyName);
+			} catch (err) {
+				this.$logger.trace(`Error while trying to get property ${propertyName} from ${projectDir}. Error is:`, err);
+			}
+		}
+
+		return null;
+	}
+
+	private getNativeScriptPropertyName(propertyName: string) {
+		return `${this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE}.${propertyName}`;
 	}
 
-	private loadProjectFile(): void {
-		assert.ok(this.projectFilePath, "Initialize method of projectDataService is not called.");
+	private getPropertyValueFromJson(jsonData: any, dottedPropertyName: string): any {
+		const props = dottedPropertyName.split(".");
+		let result = jsonData[props.shift()];
 
-		if (!this.$fs.exists(this.projectFilePath)) {
-			this.$fs.writeJson(this.projectFilePath, {
-				"description": "NativeScript Application",
-				"license": "SEE LICENSE IN <your-license-filename>",
-				"readme": "NativeScript Application",
-				"repository": "<fill-your-repository-here>"
-			});
+		for (let prop of props) {
+			result = result[prop];
 		}
 
+		return result;
+	}
+
+	private setValue(projectDir: string, key: string, value: any): void {
+		const projectFileInfo = this.getProjectFileData(projectDir);
+
+		const props = key.split(".");
+		let data: any = projectFileInfo.projectData;
+		let currentData = data;
+
+		_.each(props, (prop, index: number) => {
+			if (index === (props.length - 1)) {
+				currentData[prop] = value;
+			} else {
+				currentData[prop] = currentData[prop] || Object.create(null);
+			}
+
+			currentData = currentData[prop];
+		});
+
+		this.$fs.writeJson(projectFileInfo.projectFilePath, data, projectFileInfo.indent);
+	}
+
+	private removeProperty(projectDir: string, propertyName: string): void {
+		const projectFileInfo = this.getProjectFileData(projectDir);
+		let data: any = projectFileInfo.projectData;
+		let currentData = data;
+		const props = propertyName.split(".");
+		const propertyToDelete = props.splice(props.length - 1, 1)[0];
+
+		_.each(props, (prop, index: number) => {
+			currentData = currentData[prop];
+		});
+
+		delete currentData[propertyToDelete];
+		this.$fs.writeJson(projectFileInfo.projectFilePath, data, projectFileInfo.indent);
+	}
+
+	private getProjectFileData(projectDir: string): IProjectFileData {
+		const projectFilePath = path.join(projectDir, this.$staticConfig.PROJECT_FILE_NAME);
+
 		// Detect indent and use it later to write JSON.
-		let projectFileContent = this.$fs.readText(this.projectFilePath);
+		const projectFileContent = this.$fs.readText(projectFilePath);
+		const indent = projectFileContent ? this.detectIndent(projectFileContent) : "\t";
 
-		this.projectFileIndent = projectFileContent ? this.detectIndent(projectFileContent) : "\t";
+		const projectData = projectFileContent ? JSON.parse(projectFileContent) : Object.create(null);
 
-		this.projectData = projectFileContent ? JSON.parse(projectFileContent) : Object.create(null);
+		return {
+			projectData,
+			indent,
```

Comment:
I think we don't need to detect the `indent`. We will detect it here https://github.com/telerik/mobile-cli-lib/blob/master/file-system.ts#L199